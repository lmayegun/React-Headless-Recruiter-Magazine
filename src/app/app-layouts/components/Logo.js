import React from 'react';
import {Typography} from '@material-ui/core';
import {makeStyles} from '@material-ui/styles';
import clsx from 'clsx';

const useStyles = makeStyles(theme=>{
  return({
    root: {
      '& .logo-icon': {
        width     : '240px',
        height    : 'auto',
        transition: theme.transitions.create(['width', 'height'], {
            duration: theme.transitions.duration.shortest,
            easing  : theme.transitions.easing.easeInOut
        })
      },
      '& .react-badge, & .logo-text': {
        fontSize : '10px',
        transition: theme.transitions.create('opacity', {
            duration: theme.transitions.duration.shortest,
            easing  : theme.transitions.easing.easeInOut
        })
      }
    },
  });
});

const Logo = ( props ) => {
  const classes = useStyles();
  return (
    <div className={clsx(classes.root, "items-center")}>
        <img className="logo-icon" src="assets/images/logos/logo.png" alt="logo"/>

          { props.brandDescription && (
            <Typography className="text-16 ml-4 font-light logo-text" color="textPrimary">
               News & business intelligence for recruiters
            </Typography>
            )
          }

    </div>
  )
}

Logo.defaultProps = {
  brandDescription: true
}

export default Logo;
