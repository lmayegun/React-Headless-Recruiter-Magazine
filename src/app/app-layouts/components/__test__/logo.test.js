import React from 'react';
import {shallow} from 'enzyme';
import Logo from '../Logo';

test('should render Logo correctlly', () => {
  const wrapper = shallow(<Logo brandDescription={false}/>);
  expect(wrapper).toMatchSnapshot();
});
