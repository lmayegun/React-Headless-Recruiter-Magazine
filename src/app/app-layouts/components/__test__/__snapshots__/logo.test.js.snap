// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Logo correctlly 1`] = `
<div
  className="items-center"
>
  <img
    alt="logo"
    className="logo-icon"
    src="/assets/images/logos/logo.png"
  />
</div>
`;
